docs/source/image_process.mdx

```diff
@@ -0,0 +1,157 @@
+
+```py
+>>> dataset = load_dataset("food101", split="train[100:]").cast_column('image', Image(decode=False))
+```
+
+## ImageFolder
+
+You can also load your image dataset with a `ImageFolder` dataset builder without writing a custom dataloader. Your image dataset structure should look like this:
+
+```
+folder/dog/golden_retriever.png
+folder/dog/german_shepherd.png
+folder/dog/chihuahua.png
+
+folder/cat/maine_coon.png
+folder/cat/bengal.png
+folder/cat/birman.png
+```
+
+Then load your dataset by specifying `imagefolder` and the directory of your dataset in `data_dir`:
+
+```py
+>>> from datasets import load_dataset
+>>> dataset = load_dataset("imagefolder", data_dir="/path/to/folder")
+```
+
+Load remote datasets from their URLs with the `data_files` parameter:
+
+```py
+>>> dataset = load_dataset("imagefolder", data_files="https://download.microsoft.com/download/3/E/1/3E1C3F21-ECDB-4869-8368-6DEBA77B919F/kagglecatsanddogs_3367a.zip", split="train")
+```
+
+## Data augmentation
+
+Adding data augmentations to a dataset is common to prevent overfitting and achieve better performance. You can use any library or package you want to apply the augmentations. This guide will use the transforms from [torchvision](https://pytorch.org/vision/stable/transforms.html).
+
+Add the [`ColorJitter`](https://pytorch.org/vision/stable/transforms.html#torchvision.transforms.ColorJitter) transform to change the color properties of the image randomly:
+
+```py
+>>> from torchvision.transforms import Compose, ColorJitter, ToTensor
+
+>>> jitter = Compose(
+... [
+...  ColorJitter(brightness=0.25, contrast=0.25, saturation=0.25, hue=0.7),
+...  ToTensor(),
+...  ]
+... )
+```
+
+Create a function to apply the `ColorJitter` transform to an image:
+
+```py
+>>> def transforms(examples):
+...     examples["pixel_values"] = [jitter(image.convert("RGB")) for image in examples["image"]]
+...     return examples
+```
+
+Then you can use the [`~Dataset.set_transform`] function to apply the transform on-the-fly to consume less disk space. Use this function if you only need to access the examples once:
+
+```py
+>>> dataset.set_transform(transforms)
+```
+
+Now visualize the results of the `ColorJitter` transform:
+
+```py
+>>> import numpy as np
+>>> import matplotlib.pyplot as plt
+
+>>> img = dataset[0]["pixel_values"]
+>>> plt.imshow(img.permute(1, 2, 0))
+```
+
+![image_process_jitter](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/image_process_jitter.png)
+
+## Map
+
+[`~Dataset.map`] can also apply transforms over an entire dataset and it also generates a cache file.
+
+Create a simple [`Resize`](https://pytorch.org/vision/stable/generated/torchvision.transforms.Resize.html) function:
+
+```py
+>>> from torchvision.transforms import Resize
+
+>>> resize = Resize([100,100])
+>>> def transforms(examples):
+...     examples["pixel_values"] = [resize(image.convert("RGB")) for image in examples["image"]]
+...     return examples
```

Comment:
Let's avoid importing `torchvision` here to not to get angry the TF/Jax folks that want to run this example:
```suggestion
>>> def transforms(examples):
...     examples["pixel_values"] = [image.convert("RGB").resize((100, 100)) for image in examples["image"]]
...     return examples
```